packages/gatsby/src/internal-plugins/functions/gatsby-node.ts

```diff
@@ -43,6 +43,55 @@ interface IGlobPattern {
   globPattern: string
 }
 
+// During development, we lazily compile functions only when they're requested.
+// Here we keep track of which functions have been requested so are "active"
+const activeDevelopmentFunctions = new Set<IFunctionData>()
+let activeEntries = {}
+
+async function _restartWebpack(
+  compiler,
+  config,
+  callback
+): Promise<webpack.Compiler> {
+  return await new Promise(resolve =>
+    compiler.close(async () => {
+      compiler = webpack(config).watch({}, callback)
+      resolve(compiler)
+    })
+  )
+}
+
+// Throttle restarting the compilier to once every 500ms so we
+// don't overwhelm the CI server during tests
+const restartWebpack = _.throttle(_restartWebpack, 500)
```

Comment:
Can you explain why this is necessary? This looks very suspicious

Reply:
if you hit many uncompiled functions all at once, it triggers restarting the webpack server for each one, which could potentially overwhelm the server. This ensures webpack restarts only once every 500ms.

Reply:
Hmmm better actually would just be to run tests serially with `--runInBand` — even with this, some tests are randomly exceeding the 5000ms default test limit. I didn't like either writing code just to help the tests pass.